homeassistant/components/broadlink/__init__.py

```diff
@@ -1,31 +1,53 @@
 """The Broadlink integration."""
 from dataclasses import dataclass, field
 
+import psutil
+
 from .const import DOMAIN
 from .device import BroadlinkDevice
+from .discovery import BroadlinkDiscovery
+from .helpers import get_broadcast_addrs
 
 
 @dataclass
 class BroadlinkData:
-    """Class for sharing data within the Broadlink integration."""
+    """Class for sharing data in the Broadlink integration."""
 
+    discovery: BroadlinkDiscovery = None
+    config: dict = field(default_factory=dict)
     devices: dict = field(default_factory=dict)
     platforms: dict = field(default_factory=dict)
 
 
 async def async_setup(hass, config):
     """Set up the Broadlink integration."""
-    hass.data[DOMAIN] = BroadlinkData()
+    data = hass.data[DOMAIN] = BroadlinkData()
+    nics = await hass.async_add_executor_job(psutil.net_if_addrs)
+    data.config["broadcast_addrs"] = get_broadcast_addrs(nics)
```

Comment:
I think it might make sense to start discovery here instead of `async_setup_entry`

Reply:
What if the user removes all entries and adds them again? This method will not be called again and the discovery will be disabled until the next restart in this side case.

Reply:
I guess nothing is going to trigger discovery to start automatically since there isn't anything to load the config flow AFAICT.

Reply:
Maybe it makes sense to start it both places and add dhcp discovery

Reply:
I can do a quick PR to add dhcp support

Reply:
https://github.com/home-assistant/core/pull/48408

Reply:
If you start the discovery in `async_setup` as soon as one of them is found by dhcp (it passive so it can take a few hours to renew), you should be able to find the rest of them (assuming multiple) right away.

Reply:
That would be nice. Broadlink MAC addresses usually start with "34:EA:34", so it may be a good way to catch at least the official devices. There are many devices from different brands with Broadlink-DNA, we should leave their MACs out of DHCP as not all devices from these brands have Broadlink-DNA.

Reply:
I included the list of ones I had at home in #48408. Most of mine actually start with `B4:43:0D` 🤷‍♂️ .  If it can't cannot or isn't actually a broadlink device, the flow will abort automatically.